ehitchhiking-ui/src/app/shared/components/blacklist/blacklist.component.ts

```diff
@@ -1,28 +1,32 @@
 import {Component, OnInit} from '@angular/core';
-import {BLACKLIST_DRIVERS, BLACKLIST_PASSENGERS} from './blacklist-users';
 import {MatDialogRef} from '@angular/material';
 import {LoaderSize} from '@shared/enums/pre-loader-sizes';
 import {NoDataSize} from '@shared/enums/no-data-sizes';
+import {User} from '@shared/models/user';
+import {BlackListApiService} from '@shared/api-services/black-list-api.service';
+import {CUR_USER} from '@shared/components/blacklist/blacklist-users';
 
 @Component({
 	selector: 'app-blacklist',
 	templateUrl: './blacklist.component.html',
 	styleUrls: ['./blacklist.component.sass'],
+	providers: [BlackListApiService],
 })
 export class BlacklistComponent implements OnInit {
-	blacklistDriverArray = [];
-	blacklistPassengerArray = [];
+	blacklistDriverArray: User[] = [];
+	blacklistPassengerArray: User[] = [];
+	curUser: User = CUR_USER;
 	loaderSize: LoaderSize = LoaderSize.Large;
 	noDataSize: NoDataSize = NoDataSize.Small;
 	noDataMessage = 'No users!';
 	noDataIconName = 'accessibility';
 	loadingDrives = true;
 	loadingPassengers = true;
-	constructor(public dialogRef: MatDialogRef<BlacklistComponent>) {}
+	constructor(public dialogRef: MatDialogRef<BlacklistComponent>, private apiBlacklistService: BlackListApiService) {}
 
 	ngOnInit() {
-		this.blacklistDriverArray = BLACKLIST_DRIVERS;
-		this.blacklistPassengerArray = BLACKLIST_PASSENGERS;
+		this.apiBlacklistService.doGetDrivers(this.curUser);
```

Comment:
pls subscribe on Observable and change loading status

Reply:
done